### PR TITLE
[WIP] Resolve local meta node deps

### PR DESCRIPTION
**Pull request by MovieStoreGuy**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/AlexsJones/kepler/commands/palette"
 	"github.com/AlexsJones/kepler/commands/storage"
 	"github.com/AlexsJones/kepler/commands/submodules"
+	"github.com/Alexsjones/kepler/commands/meta"
```
Comment (AlexsJones) [inline main.go]: Casing :(

File: commands/meta/node.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	"github.com/AlexsJones/kepler/commands/submodules"
+)
+
+// Information ...
+type Information struct {
+	Projects map[string]*Node.PackageJSON
+}
+
+// NewInformation creates a struct containing information about the meta repo
+func NewInformation() (*Information, error) {
```
Comment (AlexsJones) [inline commands/meta/node.go]: Weird weird name

**Review by AlexsJones — CHANGES_REQUESTED** [review]
A single casing error might cause issue, overall I think its great but lets try to consolidate using two node command palettes, my gut feeling is move all node specific functions into the commands/node and have the meta repo rerefences it

File: commands/node/node.go [hunk 3]
```diff
@@ -8,15 +8,16 @@ import (
...
 )
 
 //AddCommands for the node module
 func AddCommands(cli *cli.Cli) {
 
 	cli.AddCommand(command.Command{
-		Name: "npm",
-		Help: "npm command palette",
+		Name: "node",
+		Help: "node command palette",
```
Comment (MovieStoreGuy) [inline commands/node/node.go:20]: I have renamed it to node as we are doing more than what npm can do

File: commands/node/commands.go [hunk 4]
```diff
@@ -86,5 +90,72 @@ func fixLinks(subPath string, filename string, prefix string, target string, sho
...
+	Projects := make(map[string]*PackageJSON)
+	submodules.LoopSubmodules(func(sub *git.Submodule) {
+		filepath := path.Join(sub.Config().Path, "package.json")
+		if _, node := os.Stat(filepath); !os.IsNotExist(node) {
+			b, err := ioutil.ReadFile(filepath)
+			if err != nil {
+				return
+			}
+			var p PackageJSON
+			json.Unmarshal(b, &p)
```
Comment (AlexsJones) [inline commands/node/commands.go:106]: error handling

File: commands/node/commands.go [hunk 5]
```diff
@@ -86,5 +90,72 @@ func fixLinks(subPath string, filename string, prefix string, target string, sho
...
+// ResolveLocalDependancies will explore (via some graph expansion)
+// once it is completed, it will return the list of the required
+// pacakages otherwise, return an informative error
+func ResolveLocalDependancies(project string) ([]string, error) {
+	LocalPackages, err := LocalNodeModules()
+	if err != nil {
+		return []string{}, err
+	}
+	if _, exists := LocalPackages[project]; !exists {
+		return nil, fmt.Errorf("%s does not exists", project)
```
Comment (AlexsJones) [inline commands/node/commands.go]: exist

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**